### ISSUE TITLE
Optimize battery life with DS18B20

### ISSUE THREAD

**Issue by Toshik** [mysensors/NodeManager#72]
For now NodeManager uses "standard" DS18B20 data retrieval:
```
  _sensors->requestTemperatures();
  float temperature = _sensors->getTempCByIndex(_index);
```
So, requestTemperatures() uses just delay/loop to wait required time for data conversion from DS18B20. And this time is up to 750ms

My proposal is use async mode, as I do in my battery powered projects: 
disable delay in requestTemperatures() by calling ```_sensors->setWaitForConversion(false)```. Then save some power by:
```
  // request temperature measurement
  _sensors->requestTemperatures();
  // sleep in some low power mode, while DS18B20 calculates temperature
  sleep(750); // or calculate exact wait time according to current DS18B20 resolution
  float temperature = _sensors->getTempCByIndex(_index);
```

That way we can make battery work little bit longer

**Comment by user2684** [comment]
Thanks for reporting this! I'll make it as an option for the next release

**Comment by luizrrocha** [comment]
Hello Toshik/user2684.

I do like this:
  // Fetch temperatures from Dallas sensors
  sensors.requestTemperatures();

  // query conversion time and sleep until conversion completed
  int16_t conversionTime = sensors.millisToWaitForConversion(sensors.getResolution());
  sleep(conversionTime);

This way I do not have to worry about finding the sleep time :-)

**Comment by user2684** [comment]
I've added `void setSleepDuringConversion(bool value);` to the SensorDs18b20 class, default is false to avoid creating issues with non-sleeping nodes.

For those using the default DallasTemperature library (not the one in the MySensors bundle), millisToWaitForConversion() is declared as private so has to be moved into public in DallasTemperature.h to have this compiling correctly.
The update will be made available in v1.5-dev2 that will be pushed to github later today.
